src/librustc/ty/sty.rs

```diff
@@ -77,6 +77,17 @@ impl BoundRegion {
             _ => false,
         }
     }
+
+    /// When canonicalizing, we replace unbound inference variables and free
+    /// regions with anonymous late bound regions. This method asserts that
+    /// we have an anonymous late bound region, which hence may refer to
+    /// a canonical variable.
+    pub fn as_bound_var(&self) -> BoundVar {
```

Comment:
Can we call this `assert_bound_var()`? I forget how consistently we use this convention, but I like it (since you are not returning an `Option` here...)